recipes-sota/config/files/fuse_actions.sh

```
@@ -0,0 +1,354 @@
+  fuse-val9: $fuse_val_9
+  fuse-val10: $fuse_val_10
+  fuse-val11: $fuse_val_11
+  fuse-val12: $fuse_val_12
+  fuse-val13: $fuse_val_13
+  fuse-val14: $fuse_val_14
+  fuse-val15: $fuse_val_15
+  fuse-val16: $fuse_val_16
+EOF
+    fi
+
+    local fuse_close
+    fuse_close=$(FW_PRINTENV fuse_val_close | SED -Ee "s/\s*fuse_val_close=(.*)$/\1/")
+    if [ "$fuse_close" = 0 ]; then
+        log "Device is currently open"
+        local prog_close="false"
+    elif [ "$fuse_close" = 1 ]; then
+        log "Device is currently closed"
+        local prog_close="true"
+    else
+        # If we can't determine the closing state from U-Boot then just assume it's open
+        log "Unable to determine Device closed state, will assume device is open"
+        local prog_close="false"
+    fi
+
+cat << EOF >> "$SECONDARY_FIRMWARE_PATH"
+  fuse-close: $prog_close
+EOF
+}
+
+# Creates target_name file if it does not already exist
+create_target_file () {
+    local fuse_dir
+    fuse_dir=$(DIRNAME "$SECONDARY_FIRMWARE_PATH")
+
+    target_file="${fuse_dir}/target_name"
+    if [ ! -e "${target_file}" ]; then
+        log "Generating target_name file"
+	echo "fuse.yml" > "$target_file"
+    fi
+}
+
+# Create firmware file from information available from U-boot if file does not exist
+do_create_firmware() {
+    log "Generating firmware file based on information in U-Boot"
+
+    create_yaml_content
+    create_target_file
+    return 0
+}
+
+do_install() {
+    before_dying 'on_install_failed'
+
+    check_install_vars
+    check_target_sha256
+
+    # Define variables early as "local" will mask the true exit code of the functions
+    local fuse_val_list fuse_close
+
+    fuse_val_list=$(parse_fuse_vals) || die "Aborting update unable to parse fuse values"
+    fuse_close=$(parse_fuse_close) || die "Aborting update unable to parse fuse values"
+
+    # Set U-Boot environment variables for uEnv.txt
+    # TODO: Do we need bootcount & upgrade_available for this type of update?
```

Answering the question here: I don't think errors with fusing should ever trigger an OS rollback. Doing so would most likely cause more harm than good since it would confuse Aktualizr (as it doesn't know about any OS update) and it wouldn't solved the fusing problem as far as I can tell.

Also, given the nature of the operation, I don't think we should have any kind of retry. Instead IMO we should report an error in failure cases and let users decide if they want to retry.

---

Removed in V2